src/System.Private.CoreLib/shared/System/Decimal.DecCalc.cs

```diff
@@ -2473,7 +2462,7 @@ internal static void InternalRound(ref DecCalc d, uint scale, RoundingMode mode)
                         ulong tmp = d.Low64;
                         if (tmp == 0)
                         {
-                            if (mode <= RoundingMode.Truncate)
+                            if (mode <= MidpointRounding.ToZero)
```

Comment:
:exclamation: This is not correct - `Truncate` and `MidpointRounding.ToZero` are two completely different things: `Truncate(1.9m) == 1` while `MidpointRounding.ToZero(1.9m) == 2`